### DISCUSSION TITLE
No data from TuYa WSD500A temperature & humidity sensor

### DISCUSSION THREAD

**Discussion by martas99** [Koenkk/zigbee2mqtt#16366] [Devices]
Hi, may I ask for help with WSD500A TuYa Temperature and humidity sensor? It pairs fine, exposes temperature, humidity and battery labels but provides no data on them (part of state.json):
```
    "0xa4c138081540b3c5": {
        "linkquality": 252
    }
```
Part of configuration.yaml:
```
  '0xa4c138081540b3c5':
    friendly_name: Teplomer
    homeassistant: {}
    legacy: false
    optimistic: true
```
I am running Zigbee2MQTT 1.29.2-1. The model ID is 'TS0601', manufacturerName '_TZE200_9yapgbuv'', model 'WSD500A', vendor 'TuYa'. Log entry is
```
info  2023-01-26 14:46:25: Successfully configured '0xa4c138081540b3c5'
info  2023-01-26 14:46:26: Successfully interviewed '0xa4c138081540b3c5', device has successfully been paired
info  2023-01-26 14:46:26: Device '0xa4c138081540b3c5' is supported, identified as: TuYa Temperature & humidity sensor (WSD500A)
```
Thanks

**Comment by Vorlex** [comment]
I've got the same problem: the same device and the same issue.

**Comment by cortlieb** [comment]
Same here! I have 3 of these sensors. Same behavior for all of them.

**Comment by hometheater13** [comment]
Have the same issue , i have 2 sensor  .  
 Zigbee2MQTT   ver 1.30.2-1   .
 Zigbee Model
TS0601
Zigbee Manufacturer
_TZE200_9yapgbuv
Model
[WSD500A](https://www.zigbee2mqtt.io/devices/WSD500A.html#tuya-wsd500a)

**Comment by Koenkk** [comment]
Could you check if the issue is fixed with the following external converter: 

```js
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const ota = require('zigbee-herdsman-converters/lib/ota');
const tuya = require('zigbee-herdsman-converters/lib/tuya');
const utils = require('zigbee-herdsman-converters/lib/utils');
const globalStore = require('zigbee-herdsman-converters/lib/store');
const e = exposes.presets;
const ea = exposes.access;

const definition = {
    fingerprint: [{modelID: 'TS0601', manufacturerName: '_TZE200_bq5c8xfe'},
        {modelID: 'TS0601', manufacturerName: '_TZE200_bjawzodf'},
        {modelID: 'TS0601', manufacturerName: '_TZE200_qyflbnbj'},
        {modelID: 'TS0601', manufacturerName: '_TZE200_9yapgbuv'},
        {modelID: 'TS0601', manufacturerName: '_TZE200_zl1kmjqx'}],
    model: 'TS0601_temperature_humidity_sensor',
    vendor: 'TuYa',
    description: 'Temperature & humidity sensor',
    fromZigbee: [fz.tuya_temperature_humidity_sensor],
    toZigbee: [],
    exposes: (device, options) => {
        const exps = [e.temperature(), e.humidity(), e.battery()];
        if (!device || device.manufacturerName === '_TZE200_qyflbnbj') {
            exps.push(e.battery_low());
            exps.push(exposes.enum('battery_level', ea.STATE, ['low', 'middle', 'high']).withDescription('Battery level state'));
        }
        exps.push(e.linkquality());
        return exps;
    },
};

module.exports = definition;

```
- save this as file next to `configuration.yaml` as `ext_converter.js`
- add it to `configuration.yaml`:
```yaml
external_converters:
  - ext_converter.js
```
- start z2m, check if issue is fixed

**Reply by Vorlex** [reply]
~~Unfortunately, for me, it's still the same: temp, humidity, and battery are N/A.
Although, the device now has another image.
And the interview process took much longer this time.~~

Update: it started to work after I replaced the stock battery. Thanks.

**Reply by hometheater13** [reply]
> Could you check if the issue is fixed with the following external converter:
> 
> ```js
> const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
> const tz = require('zigbee-herdsman-converters/converters/toZigbee');
> const exposes = require('zigbee-herdsman-converters/lib/exposes');
> const reporting = require('zigbee-herdsman-converters/lib/reporting');
> const extend = require('zigbee-herdsman-converters/lib/extend');
> const ota = require('zigbee-herdsman-converters/lib/ota');
> const tuya = require('zigbee-herdsman-converters/lib/tuya');
> const utils = require('zigbee-herdsman-converters/lib/utils');
> const globalStore = require('zigbee-herdsman-converters/lib/store');
> const e = exposes.presets;
> const ea = exposes.access;
> 
> const definition = {
>     fingerprint: [{modelID: 'TS0601', manufacturerName: '_TZE200_bq5c8xfe'},
>         {modelID: 'TS0601', manufacturerName: '_TZE200_bjawzodf'},
>         {modelID: 'TS0601', manufacturerName: '_TZE200_qyflbnbj'},
>         {modelID: 'TS0601', manufacturerName: '_TZE200_9yapgbuv'},
>         {modelID: 'TS0601', manufacturerName: '_TZE200_zl1kmjqx'}],
>     model: 'TS0601_temperature_humidity_sensor',
>     vendor: 'TuYa',
>     description: 'Temperature & humidity sensor',
>     fromZigbee: [fz.tuya_temperature_humidity_sensor],
>     toZigbee: [],
>     exposes: (device, options) => {
>         const exps = [e.temperature(), e.humidity(), e.battery()];
>         if (!device || device.manufacturerName === '_TZE200_qyflbnbj') {
>             exps.push(e.battery_low());
>             exps.push(exposes.enum('battery_level', ea.STATE, ['low', 'middle', 'high']).withDescription('Battery level state'));
>         }
>         exps.push(e.linkquality());
>         return exps;
>     },
> };
> 
> module.exports = definition;
> ```
> 
> * save this as file next to `configuration.yaml` as `ext_converter.js`
> * add it to `configuration.yaml`:
> 
> ```yaml
> external_converters:
>   - ext_converter.js
> ```
> 
> * start z2m, check if issue is fixed

Hi

tested , after few hours get the Temperature and Humidity data , battery still missing  .|




`Zigbee2MQTT:info  2023-03-09 08:58:50: MQTT publish: topic 'homeassistant/sensor/0xa4c1385058b7740b/temperature/config', payload '{"availability":[{"topic":"zigbee2mqtt/bridge/state"},{"topic":"zigbee2mqtt/LR AV2 Temp/availability"}],"availability_mode":"all","device":{"identifiers":["zigbee2mqtt_0xa4c1385058b7740b"],"manufacturer":"TuYa","model":"Temperature & humidity sensor (TS0601_temperature_humidity_sensor)","name":"LR AV2 Temp"},"device_class":"temperature","enabled_by_default":true,"name":"LR AV2 Temp temperature","state_class":"measurement","state_topic":"zigbee2mqtt/LR AV2 Temp","unique_id":"0xa4c1385058b7740b_temperature_zigbee2mqtt","unit_of_measurement":"°C","value_template":"{{ value_json.temperature }}"}'
Zigbee2MQTT:info  2023-03-09 08:58:50: MQTT publish: topic 'homeassistant/sensor/0xa4c1385058b7740b/humidity/config', payload '{"availability":[{"topic":"zigbee2mqtt/bridge/state"},{"topic":"zigbee2mqtt/LR AV2 Temp/availability"}],"availability_mode":"all","device":{"identifiers":["zigbee2mqtt_0xa4c1385058b7740b"],"manufacturer":"TuYa","model":"Temperature & humidity sensor (TS0601_temperature_humidity_sensor)","name":"LR AV2 Temp"},"device_class":"humidity","enabled_by_default":true,"name":"LR AV2 Temp humidity","state_class":"measurement","state_topic":"zigbee2mqtt/LR AV2 Temp","unique_id":"0xa4c1385058b7740b_humidity_zigbee2mqtt","unit_of_measurement":"%","value_template":"{{ value_json.humidity }}"}'
Zigbee2MQTT:info  2023-03-09 08:58:50: MQTT publish: topic 'homeassistant/sensor/0xa4c1385058b7740b/battery/config', payload '{"availability":[{"topic":"zigbee2mqtt/bridge/state"},{"topic":"zigbee2mqtt/LR AV2 Temp/availability"}],"availability_mode":"all","device":{"identifiers":["zigbee2mqtt_0xa4c1385058b7740b"],"manufacturer":"TuYa","model":"Temperature & humidity sensor (TS0601_temperature_humidity_sensor)","name":"LR AV2 Temp"},"device_class":"battery","enabled_by_default":true,"entity_category":"diagnostic","name":"LR AV2 Temp battery","state_class":"measurement","state_topic":"zigbee2mqtt/LR AV2 Temp","unique_id":"0xa4c1385058b7740b_battery_zigbee2mqtt","unit_of_measurement":"%","value_template":"{{ value_json.battery }}"}'
Zigbee2MQTT:info  2023-03-09 08:58:50: MQTT publish: topic 'homeassistant/sensor/0xa4c1385058b7740b/linkquality/config', payload '{"availability":[{"topic":"zigbee2mqtt/bridge/state"},{"topic":"zigbee2mqtt/LR AV2 Temp/availability"}],"availability_mode":"all","device":{"identifiers":["zigbee2mqtt_0xa4c1385058b7740b"],"manufacturer":"TuYa","model":"Temperature & humidity sensor (TS0601_temperature_humidity_sensor)","name":"LR AV2 Temp"},"enabled_by_default":false,"entity_category":"diagnostic","icon":"mdi:signal","name":"LR AV2 Temp linkquality","state_class":"measurement","state_topic":"zigbee2mqtt/LR AV2 Temp","unique_id":"0xa4c1385058b7740b_linkquality_zigbee2mqtt","unit_of_measurement":"lqi","value_template":"{{ value_json.linkquality }}"}'
Zigbee2MQTT:info  2023-03-09 08:58:50: MQTT publish: topic 'homeassistant/sensor/0xa4c1385058b7740b/last_seen/config', payload '{"availability":[{"topic":"zigbee2mqtt/bridge/state"},{"topic":"zigbee2mqtt/LR AV2 Temp/availability"}],"availability_mode":"all","device":{"identifiers":["zigbee2mqtt_0xa4c1385058b7740b"],"manufacturer":"TuYa","model":"Temperature & humidity sensor (TS0601_temperature_humidity_sensor)","name":"LR AV2 Temp"},"device_class":"timestamp","enabled_by_default":false,"entity_category":"diagnostic","icon":"mdi:clock","name":"LR AV2 Temp last seen","state_topic":"zigbee2mqtt/LR AV2 `



![image](https://user-images.githubusercontent.com/64803530/223945506-2dd488d2-7eff-420d-93ca-102738b18f19.png)

**Reply by nakedfoot** [reply]
Hello,

Tried the same converter, changed the battery, but still no data.
I got 4 sensors (2 different vendors, but same model "_TZE200_9yapgbuv").
I'm going crazy...
tried with zigbee2mqtt and zigbee2mqtt_edge

Do i have to manually update herdsman-converters (don't know how) ?

Thx.

**Reply by Koenkk** [reply]
If you go to the "State" tab instead of the "Exposes" tab, do you see a `battery_level`?

**Reply by nakedfoot** [reply]
No. Only 
{ 
"linkquality": 255
} 

I have an old Tuya GateWay somewhere in my desk. 
Maybe i can try to make my sensor work with it and send datapoints here.

In Zigbee2MQTT log, i only see this message :
type 'attributeReport', cluster 'genbasic', data '{"65506":56, "65508":1, "appVersion":72} from endpoint 1 with groupID null

Nothing more for the sensor (no json data or anything else)

Thx.

**Reply by Koenkk** [reply]
Can you try to:

- force remove the device through z2m frontend
- restart z2m
- re-pair device
-

**Reply by nakedfoot** [reply]
U R my savior.
It worked. ;-)

But I had to stop an start Z2M (just "restart" didn't worked).

**Reply by Heronimos** [reply]
Nope, I don't get any value.  Not even after stop / start or even complete restart between the force delete and adding it again

After a time, 1 is working, the other 2 don't.  Don't get it, look like the interview is not complete or something..

**Reply by Heronimos** [reply]
![afbeelding](https://user-images.githubusercontent.com/89809494/229279250-256cbb4d-4e15-41be-98df-dbeb221f31a1.png)

Everything is the same on every tab, only state differs

![afbeelding](https://user-images.githubusercontent.com/89809494/229279301-31b18e44-63f9-4f2f-950f-ec61d5dceb8b.png)

Where do I start my search?

**Comment by martas99** [comment]
Yeap, it works! Thank you, @hometheater13!
My units have different icon and battery is not shown in %, but I don't care...
![obrazek](https://user-images.githubusercontent.com/114129681/224054739-1e38ef19-2446-49c5-b527-afcbab0760b4.png)

**Comment by Koenkk** [comment]
Probably this particular version support just `battery_low`. To double check, can someone provide me the debug log when running z2m with this sensor for +- 24 hours? I'm interested in messages like: `zigbee-herdsman-converters:maa_tuya_temp_sensor: NOT RECOGNIZED`

**Reply by Heronimos** [reply]
Lines like this:
`
warn  2023-04-01 12:23:16: zigbee-herdsman-converters:maa_tuya_temp_sensor: NOT RECOGNIZED DP #9 with data {"dp":9,"datatype":4,"data":{"type":"Buffer","data":[0]}}`

I can send complete log's, but don't know who to filter. 

As far I can see the above line is after connect with the one sensor that he does recognize.
The one that he doesn't does:

```
debug 2023-04-01 13:23:03: Device '0xa4c138fb256ac713' announced itself
info  2023-04-01 13:23:03: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0xa4c138fb256ac713","ieee_address":"0xa4c138fb256ac713"},"type":"device_announce"}'
debug 2023-04-01 13:23:03: Received Zigbee message from 'Plug Z2', type 'attributeReport', cluster 'seMetering', data '{"currentSummDelivered":[0,13]}' from endpoint 1 with groupID 0
debug 2023-04-01 13:23:07: Received Zigbee message from '0xa4c138fb256ac713', type 'read', cluster 'genTime', data '["localTime"]' from endpoint 1 with groupID 0
```

debug 2023-04-01 13:23:08: Received Zigbee message from '0xa4c138fb256ac713', type 'attributeReport', cluster 'genBasic', data '{"65506":56,"65508":1,"appVersion":72}' from endpoint 1 with groupID 0

**Comment by martas99** [comment]
more than 50x since yesterday noon:
`warn  2023-03-10 08:05:43: zigbee-herdsman-converters:maa_tuya_temp_sensor: NOT RECOGNIZED DP #9 with data {"dp":9,"datatype":4,"data":{"type":"Buffer","data":[0]}}`

**Reply by Koenkk** [reply]
Do you maybe have a tuya gateway? Then we can easily found out what it means: https://www.zigbee2mqtt.io/advanced/support-new-devices/03_find_tuya_data_points.html

**Reply by martas99** [reply]
No tuya GW available, I'm affraid...

**Reply by Jonas-Meyer97** [reply]
Is this helpful? Sensor: TS0601 _TZE200_9yapgbuv 
![hum](https://user-images.githubusercontent.com/5890132/226123374-6c25fcd6-6e71-4950-a47e-b23039f76fb0.PNG)
![unit](https://user-images.githubusercontent.com/5890132/226123375-e762cdf7-90c1-464a-bfed-fa58ea8f055c.PNG)
![battery](https://user-images.githubusercontent.com/5890132/226123376-7520d489-c4ff-466e-86d7-a3532ffce7fa.PNG)
![temp](https://user-images.githubusercontent.com/5890132/226123378-18a42915-d897-4d0d-9094-e47f37e198f2.PNG)

**Comment by mvadu** [comment]
@Koenkk looks like this is not `WSD500A` sensor. I have few of them as well, and they run off of 2 AAA batteries. This one is marketed as  `ZTH02` and uses a CR2032 battery. Few sellers even mention "[Zigbee2MQTT is not supported.](https://a.aliexpress.com/_m0kPVcG)" in their description. Inside of the device has ZTH02 printed on the PCB as well. 
![image](https://github.com/Koenkk/zigbee2mqtt/assets/8618235/1252a7fa-f246-4c0d-b87f-b34c4515320a)

Looking at source code, I took the `ZTH01` converter and applied like below and I am able to get the data correctly.

``` 
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const ota = require('zigbee-herdsman-converters/lib/ota');
const tuya = require('zigbee-herdsman-converters/lib/tuya');
const utils = require('zigbee-herdsman-converters/lib/utils');
const globalStore = require('zigbee-herdsman-converters/lib/store');
const e = exposes.presets;
const ea = exposes.access;

const definition = {
    fingerprint: [
        { modelID: 'TS0601', manufacturerName: '_TZE200_9yapgbuv' },
    ],
    model: 'ZTH02',
    vendor: 'TuYa',
    description: 'Temperature and humidity sensor',
    fromZigbee: [tuya.fz.datapoints, tuya.fz.gateway_connection_status],
    toZigbee: [tuya.tz.datapoints],
    configure: tuya.configureMagicPacket,
    exposes: [e.temperature(), e.humidity(), tuya.exposes.batteryState(), e.battery_low()],
    meta: {
        tuyaDatapoints: [
            [1, 'temperature', tuya.valueConverter.divideBy10],
            [2, 'humidity', tuya.valueConverter.raw],
            [3, 'battery_state', tuya.valueConverter.batteryState],
            [9, 'temperature_unit', tuya.valueConverter.raw], //This DP is not properly supported by the device
        ],
    },
};

module.exports = definition;
```

```
{
    "last_seen": "2023-05-26T20:37:29+00:00",
    "linkquality": 160,
    "battery": null,
    "humidity": 60,
    "temperature": 21.4,
    "battery_level": "high",
    "battery_low": false,
    "battery_state": "high",
    "temperature_unit": 0
}
```

I am not used to z2m repo enough to add a new device but can take this is up if you are good with adding a new device.

**Reply by Koenkk** [reply]
Does the `temperature_unit` work?

**Reply by mvadu** [reply]
I have seen it be 0 always. Is there a way to trigger a change in unit via the UI? without ` [9, 'temperature_unit', tuya.valueConverter.raw]`line I was seeing the `NOT RECOGNIZED` error for code 9. Nothing after that line in the converter.

**Reply by Koenkk** [reply]
Publish to `zigbee2mqtt/DEVICE_ID/set` payload `{"temperature_unit": 1}`

**Reply by mvadu** [reply]
it changed in the state but the temperature itself didn't change.. so I think that comment is still valid.

**Reply by Koenkk** [reply]
Thanks, integrated the changes!

Changes will be available in the dev branch in a few hours from now. (https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html)

**Reply by mvadu** [reply]
Thank you @Koenkk. Two questions
1. Should the name be `ZTH02` since that is what device is sold as.
2. If answer to above is yes, should we add a documentation page for zth02 so the ui can then link to it.

**Reply by Koenkk** [reply]
I will be named `ZTH02`, docs will be automatically generated with the next release.

**Reply by saper-2** [reply]
Hi,

I got today this ```ZTH02``` sensor.

Custom converter by @mvadu - do not work, I get ```null``` (PCB marking is exactly that same in my).

Custom converter by @Koenkk - **works** in 90%, except there is no battery level (for now - let's wait few h), adn reports device as TS601 and use TS0601 image too.


Photo for ZTH02 can be found here: https://aliexpress.com/item/1005005234539150.html

FIY: I have z2m ver. 1.30.4

**Reply by mvadu** [reply]
Hi @saper-2 I had that null too.. Give it may be half an hour.. The zth02 sends messages very sparingly.

**Reply by Odilhao** [reply]
My `ZTH02` just arrived, and It was not getting the values even when using the `latest-dev` image, after adding the external converter the values started to show, thanks @mvadu

**Reply by saper-2** [reply]
Hi,

I gave it about 1/2 day, there is no detailed battery level reported, I guess it only report high/low 😑:
```JSON
{
    "battery_level": "high",
    "battery_low": false,
    "battery_state": "high",
    "humidity": 46,
    "last_seen": "2023-06-01T08:32:41.952Z",
    "linkquality": 123,
    "temperature": 24.6,
    "temperature_unit": 0,
    "battery": null
}
```

*I have another one but it get stuck on pairing, i suspect broken sensor (AHT21 if I remember right), I want try to replace it and see if this ZTH start working - if not I'll tear down it and reverse circuit to check how battery is measured.*

**Reply by mvadu** [reply]
Yes, zth02 only reports high low. What I have seen is most units using AAA batteries report voltage and level, where as units using cr2032 or similar chemistry batteries just report a status.

**Reply by saper-2** [reply]
I think you generalized this a bit too much, I have aqara and mija running on cr2332 and those report batt(V) & batt(%). Just those `ZTH02` don't report this kind of information.

A small OT: how to add to `Exposes` tab `battery_level` or `battery_state` 🤔 ?

**Reply by saper-2** [reply]
Weird thing, my ZTH02 draining battery like crazy 😑 2-3days most and it have 1.7V (might be another one broken that I got).

**Reply by mvadu** [reply]
I was thinking I am only one.. It went through 3 batteries very quickly.. Then I put forth battery (this time a genuine Panasonic) and repaired, working on same for three weeks now.. Not sure why the behavior changed..

**Comment by mvadu** [comment]
> I think you generalized this a bit too much, I have aqara and mija running on cr2332 and those report batt(V) & batt(%). Just those `ZTH02` don't report this kind of information.
> 
> A small OT: how to add to `Exposes` tab `battery_level` or `battery_state` 🤔 ?

May be.. All my other zigbee temperature sensor were AAA. The Exposes geta those entities from the config for the sensor. Latest z2m already has the sensor config so it should automatically show them if you pull Latest image and restart (and may be re pair with a fresh battery)

**Comment by saper-2** [comment]
I bumped up z2m version by `git pull` on my test machine to: **1.31.2 commit: 21f5125** (and coordinator fw too). And now converter by @Koenkk https://github.com/Koenkk/zigbee2mqtt/discussions/16366#discussioncomment-5243375 don't work:
```
saper@N10D:/opt/zigbee2mqtt $ npm start

> zigbee2mqtt@1.31.2 start
> node index.js


/opt/zigbee2mqtt/node_modules/zigbee-herdsman-converters/index.js:79
        if (converter.options) {
                      ^
TypeError: Cannot read properties of undefined (reading 'options')
    at Object.addDefinition [as addDeviceDefinition] (/opt/zigbee2mqtt/node_modules/zigbee-herdsman-converters/index.js:79:23)
    at new ExternalConverters (/opt/zigbee2mqtt/lib/extension/externalConverters.ts:15:17)
    at new Controller (/opt/zigbee2mqtt/lib/controller.ts:84:58)
    at start (/opt/zigbee2mqtt/index.js:106:18)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

While converter by @mvadu don't report anything - just `N/A`
And I see those lines in logs with this converter:
```
Zigbee2MQTT:debug 2023-06-17 19:46:39: Retrieving state of 'ZTH02' after reconnect
Zigbee2MQTT:debug 2023-06-17 19:46:40: Received Zigbee message from 'ZTH02', type 'read', cluster 'genTime', data '["localTime"]' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2023-06-17 19:46:41: Received Zigbee message from 'ZTH02', type 'attributeReport', cluster 'genBasic', data '{"65506":56,"65508":1,"appVersion":72}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2023-06-17 19:47:37: Received Zigbee message from 'RELAY_POW', type 'attributeReport', cluster 'genBasic', data '{"65506":54,"65508":0,"appVersion":70}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2023-06-17 19:47:49: Received Zigbee message from 'ZTH02', type 'attributeReport', cluster 'genBasic', data '{"65503":"\u001c� ,\u0019\u001c� ,\u0007\u0000\u0000\u0000\u0000\u0011"}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2023-06-17 19:48:04: Saving state to file /opt/zigbee2mqtt/data/state.json
```

**Reply by mvadu** [reply]
You shouldn't need any external converters anymore for zth02. Try removing them from config and restart z2m.

**Reply by saper-2** [reply]
Ok, seems my ZTH02 broken down, not only it did not report anything except LQI but it drained new battery in 10min from 3,3V to 2,88V 😑 . I do wanted to believe in those ZTH02 but **those are just a junk**, at least those 💩 didn't cost more than 3$.

**Reply by doenietzomoeilijk** [reply]
I have the same device that @mvadu posted a picture of, mine seems to have gone belly-up as well. The little LED is now blinking really quickly all the time, it only reports LQI (I've seen it report humidity and battery state at some point, but never temperature). Unfortunate. 

The battery assembly seems to be prone to letting the battery "slip" underneath it when you insert it a tad enthousiatically, perhaps this happened and it shorted the thing by contacting the switch which is right in the way, I don't know. I'll try and contact the seller, see if they can ship a different one.

I also have one of the AAA-powered ones, that one worked right away without issues.

**Comment by trwbox** [comment]
I don't know if this is the right place to put this, but I just received one of these sensors in the mail. The board is listed as ZTH02_V1.0, with date of 2022-05-27, and it is reporting a different manufacturer `_TZE204_9yapgbuv`. I've just put mvadu's external converter with it modified to change the manufacturer and it appears to be working correctly using it.
